### PR TITLE
New icon: Aegis Authenticator

### LINKED ISSUE #8385: Aegis Authenticator
### Brand Name

Aegis Authenticator

### Website

https://github.com/beemdevelopment/Aegis

### Popularity Metric

5042 GitHub stars _(as of 2023-02-25)_

#### Icon resource

~~[icon.png](https://getaegis.app/images/icon.png) from [their website](https://getaegis.app/) (at the header / footer) is the best resource they have, they don't have vector resources.~~

_EDIT 2023-02-26:_ There's now official vector by [Aegis Authenticator dev](https://github.com/alexbakker): https://alexbakker.me/u/pn8frf828n.svg

This vector is not hosted in their GitHub repo nor website.

#### Color
`#005E9D`.

Color is taken from the [icon.png](https://getaegis.app/images/icon.png).

Since the logo is using gradient, this color was taken from bottom part of the logo because it's most noticable color in my opinion.

[Website](https://getaegis.app/) in my opinion doesn't have colors that would fit the brand perfectly.

### Additional Comments

~~I have created the vectorized version of the logo from scratch based of [icon.png](https://getaegis.app/images/icon.png).~~

### Contributing

~~Yes~~. Dropped, it's all yours.

### PR DESCRIPTION
**Pull request by krisu5**
**Preview:**

aegisauthenticator

**Issue:** closes  #8385

**Similarweb rank:**
[#1,744,197](https://www.similarweb.com/website/getaegis.app/)

**GitHub stars:**
[5043](https://github.com/beemdevelopment/Aegis)

### Checklist
  - [x] I updated the JSON data in `_data/simple-icons.json`
  - [x] I optimized the icon with SVGO or SVGOMG
  - [x] The SVG `viewbox` is `0 0 24 24`

### Description
**Why did you pick the hex value?**

Since the logo is using gradient, this color was taken from bottom part of the logo because it's most noticable color in my opinion. [Website](https://getaegis.app/) in my opinion doesn't have colors that would fit the brand perfectly.

~~**Did you manually vectorize the logo?**~~

~~Yes I did, it's based of [website](https://getaegis.app/)'s [icon.png](https://getaegis.app/images/icon.png) raster file. Here's the raw AI file for the icon creation: [Aegis icon raw.zip](https://github.com/simple-icons/simple-icons/files/10831349/Aegis.icon.raw.zip) *(outdated)*.~~

_EDIT 2023-02-26:_ There's now official vector by [Aegis Authenticator dev](https://github.com/alexbakker): https://alexbakker.me/u/pn8frf828n.svg

This vector is not hosted in their GitHub repo nor website.

### CODE REVIEW

File: icons/aegisauthenticator.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Aegis Authenticator</title><path d="m16.89,20.44c.58.99-.17,2.29-1.32,2.28-.57,0-1.06-.31-1.32-.77h0s0,0,0,0c0,0,0-.02-.01-.02l-3.74-6.49c-.76-1-.02-2.49,1.24-2.47.61,0,1.13.35,1.39.86l3.77,6.64h0Zm-6.99-.04l-.37-.67c-1.04-1.76-3.51-.17-2.56,1.57l.34.64c1.08,1.74,3.51.26,2.59-1.53h0Zm.22-17.39L.23,20.38s-.05.09-.06.11c-.11.21-.17.45-.17.7,0,.84.69,1.52,1.53,1.51.53,0,1-.29,1.27-.71l7.95-13.95c.28-.46.81-.76,1.37-.7,0,0,.01,0,.02,0,.15.01.29.05.41.11.01,0,.06.03.03.01.03.01.06.03.08.05.05.03.1.06.15.09,0,0,.2.17.21.18.07.07.13.13.17.18,0,0,.1.14.16.24l7.81,13.76h0s.03.05.04.07c.27.42.74.7,1.27.7.84,0,1.52-.68,1.53-1.52,0-.26-.06-.5-.17-.71l-.02-.04h0s-7.72-13.59-7.72-13.59h0l-2.54-4.5-.08-.15h0c-.03-.05-.06-.11-.1-.17h0s0,0,0,0c-.22-.34-.61-.77-1.22-.81-.08-.01-.16,0-.23,0-.67.06-1.1.57-1.32.91"/></svg>
```
Comment (elliotwutingfeng) [inline icons/aegisauthenticator.svg:1]:
```suggestion
<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Aegis Authenticator</title><path d="M11.92 1.258c-.67.06-1.1.57-1.32.91l-.48.86L.23 20.395l-.06.112a1.512 1.512 0 0 0 1.36 2.209c.53 0 1-.29 1.27-.71l7.95-13.95c.28-.46.81-.76 1.37-.7h.02c.15.01.29.05.41.11.01 0 .06.03.03.01.03.01.06.03.08.05l.15.09s.2.17.21.18c.07.07.13.13.17.18 0 0 .1.14.16.24l7.81 13.76s.03.05.04.07c.27.42.74.701 1.27.701.84 0 1.52-.681 1.53-1.521 0-.26-.06-.5-.17-.71l-.02-.04-7.72-13.59-2.54-4.5-.08-.15c-.03-.05-.06-.11-.1-.17-.22-.34-.61-.77-1.22-.81-.08-.01-.16 0-.23 0zm-.18 11.728c-1.26-.02-2 1.471-1.24 2.471l3.74 6.49c.01 0 .01.02.01.02.26.46.75.77 1.32.77 1.15.01 1.9-1.29 1.32-2.28l.01.03-3.77-6.64a1.56 1.56 0 0 0-1.39-.86zM8.283 18.98c-1.004.015-1.965 1.14-1.312 2.336l.34.641c1.08 1.74 3.51.26 2.59-1.53v-.009l-.372-.67c-.325-.55-.79-.774-1.246-.768z"/></svg>
```

Comment (krisu5) [reply icons/aegisauthenticator.svg:1]: Thanks, it was better but needed slight tweaking for path positions.

**Comment by krisu5** [comment]
Width and X coordinate fixed, don't know how to fix other linter issues though.

**Review by elliotwutingfeng — CHANGES_REQUESTED** [review]
image

The .svg image (black) does not match when superimposed over the .ai image (blue)

**Comment by krisu5** [comment]
### Comparisons
[Raster vs. vector side-by-side](https://user-images.githubusercontent.com/3540275/221385910-4101f5ef-9529-40db-8a42-5ce13c420ce7.png)
[Vector upper layer, raster bottom layer](https://user-images.githubusercontent.com/3540275/221385911-22095835-e7e4-4ae8-911a-2ea85f3881d4.png)
[Raster to vector animation](https://user-images.githubusercontent.com/3540275/221385913-6b55130f-c692-4be5-ae8d-b75f82297c32.gif)

**Comment by elliotwutingfeng** [comment]
> ### Comparisons
> 
> [Raster vs. vector side-by-side](https://user-images.githubusercontent.com/3540275/221385910-4101f5ef-9529-40db-8a42-5ce13c420ce7.png) [Vector upper layer, raster bottom layer](https://user-images.githubusercontent.com/3540275/221385911-22095835-e7e4-4ae8-911a-2ea85f3881d4.png) [Raster to vector animation](https://user-images.githubusercontent.com/3540275/221385913-6b55130f-c692-4be5-ae8d-b75f82297c32.gif)

image

Still doesn't match.

**Comment by krisu5** [comment]
> > ### Comparisons
> > [Raster vs. vector side-by-side](https://user-images.githubusercontent.com/3540275/221385910-4101f5ef-9529-40db-8a42-5ce13c420ce7.png) [Vector upper layer, raster bottom layer](https://user-images.githubusercontent.com/3540275/221385911-22095835-e7e4-4ae8-911a-2ea85f3881d4.png) [Raster to vector animation](https://user-images.githubusercontent.com/3540275/221385913-6b55130f-c692-4be5-ae8d-b75f82297c32.gif)
> 
> image
> 
> Still doesn't match.
> 
> Was the original raster file changed?

No it wasn't, I don't get those kinds of results.

Try cropped version of icon.png: [SVG template for Inkscape](https://user-images.githubusercontent.com/3540275/221408751-1ffd483c-dedb-44ea-870f-ca62b0399c57.svg), [original cropped PNG file (just in case)](https://user-images.githubusercontent.com/3540275/221408756-603df7fb-828b-4f9f-851a-129a293af1c4.png)

Open SVG template to Inkscape, add `aegisauthenticator.svg` and set the height to `162,000` with locked aspect ratio, center vector both horizontally and vertically.

I made again minor tweaks again to it even closer to raster: [GIF animation](https://user-images.githubusercontent.com/3540275/221409054-b60f4828-809b-4edf-b209-37dd0d2845b5.gif)

**Comment by elliotwutingfeng** [comment]
Ideally if @beemdevelopment has an official svg logo it would be easier to convert it for simple-icons.

**Comment by krisu5** [comment]
> Original SVG logo now available at: https://alexbakker.me/u/pn8frf828n.svg

Okay, I update it to that.

**Review by elliotwutingfeng — CHANGES_REQUESTED** [review]

File: icons/aegisauthenticator.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg role="img" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><title>Aegis Authenticator</title><path d="M10.77 2.067.197 20.462c-.56.98.14 2.2 1.27 2.2.53 0 1.01-.28 1.27-.74L10.69 7.988c.56-.99 1.99-.99 2.55 0l7.922 13.924c.26.46.75.74 1.27.74h.1c1.131 0 1.831-1.22 1.271-2.19L13.31 2.077c-.56-.98-1.98-.99-2.54 0zm-.33 11.693c-.26.46-.26 1.01 0 1.47l3.8 6.682c.26.46.76.75 1.29.75h.02c1.141 0 1.851-1.23 1.291-2.22l-3.81-6.682c-.57-1-2.021-1-2.591 0zm-3.471 6.111c-.26.47-.28 1.029-.02 1.479l.32.56c.26.461.76.751 1.29.751h.02c1.14 0 1.85-1.23 1.29-2.22l-.32-.57c-.57-1-2.01-1-2.58 0"/></svg>
```
[the remark this answers is not in the document]
Comment (krisu5) [reply icons/aegisauthenticator.svg:1]: Thanks, but my issue with that solution is that it's wopping ~300 kb bigger.

I'm pushing at fix that only increases few kbs.

[the remark this answers is not in the document]
Comment (elliotwutingfeng) [reply icons/aegisauthenticator.svg]: > Thanks, but my issue with that solution is that it's whopping ~300 kb bigger.
> 
> I'm pushing fix that's smaller.

300kB? It reads 891 bytes on my end.

**Comment by elliotwutingfeng** [comment]
image

Current iteration (red) is slightly larger than the original (black)

File: icons/aegisauthenticator.svg (hunk 2 above)
[the remark this answers is not in the document]
Comment (krisu5) [reply icons/aegisauthenticator.svg]: I meant byte, this it *was* almost 300 bytes bigger.

**Comment by krisu5** [comment]
> [image](https://user-images.githubusercontent.com/30223404/221801357-15ea81e9-68b0-49a4-80a9-155068abef87.png)
> 
> Current iteration (red) is slightly larger than the original (black)

I don't see it, I give up.

**Comment by krisu5** [comment]
Is it now okay? Sorry for making too much trouble, I should referenced the original icon as top layer, not bottom.

If not, I will stop and leave it someone else.

**Comment by elliotwutingfeng** [comment]
image

Unfortunately, now it's quite far off.

891 bytes should be fine considering that almost half of the icons are above 1 KB. As we've found out the default precision settings in svgo cause the logo's shape to be distorted, hence the increase in precision to 4 (and subsequent increase in filesize).

**Comment by krisu5** [comment]
I don't know how it got it like that, but I give up.

**Comment by elliotwutingfeng** [comment]
> I don't know how it got it like that (I double-tested it with optimized and original in Inkscape on 24px x 24px canvas, so it cannot be like that), so I give up. I just don't have time and patience for this anymore.
> 
> EDIT: I checked it at home again, it's perfect match no matter how you look it. It's nothing close to being that off.
> 
> Am I missing something here?
> 
> > As we've found out the default precision settings in svgo cause the logo's shape to be distorted, hence the increase in precision to 4 (and subsequent increase in filesize).
> 
> This is not the info I found on CONTRIBUTING nor this thread: #5076
> 
> Advice in both of these is use default settings for SVGO / SVGOMG.

Hi, I'm very sorry for messing up the latest comparison.

I just checked the icons again. Your latest svg is a very close fit.

The idea to increase precision was given to me in an earlier pull request by one of the project members (https://github.com/simple-icons/simple-icons/pull/7836#issuecomment-1253525524)

**Pull request closed without merging**